### DISCUSSION TITLE
Help needed on interpreting taco2 dca specs vs. audio sample specs from tensorboard (gt)

### DISCUSSION THREAD

**Discussion by thorstenMueller** [coqui-ai/TTS#424] [General Q&A]
Hello.

I've trained a taco2 dca model for 360k steps and am running a fullband-melgan vocoder training for same dataset currently.
Quality differs from audio samples in tb (ground truth) and synthesized tts version. Synthesized version is much more metalic than audio sample in tb.

I've synthesized tb sample audio sentence ("_Den Verwendungszweck vorher nicht anzusagen, zeugt schon von einer gewissen Aroganz._") with several taco2 dca checkpoints but that didn't change anything. I've opened tb and tts files/specs in audacity (see screenshot) including 4 audio files / specs.
1. spec: audio sample from tb
2. spec: tts taco2 dca with checkpoint 350k
3. spec: tts taco2 dca with checkpoint 200k
4. spec: tts taco2 dca with checkpoint 360k

<img width="1341" alt="specs tb sample vs tts taco2 dca" src="https://user-images.githubusercontent.com/9558265/114365463-c5190a00-9b7a-11eb-99f8-f1ba83ec2081.png">

I could need some help on interpreting these. Is my taco2 dca model not good (specs 2,3,4)?

Tensorboard (tb) audio sample from 350k:
https://drive.google.com/file/d/1HMZEikEG40vFN8D3x-ljEQgRgo7R3e03/view?usp=sharing

tts synthesized (taco2 dca) sample with 350k checkpoint:
https://drive.google.com/file/d/13u804pEqH_u-B1RQK_ihYrNPBRGqWkrH/view?usp=sharing

Taco2 dca and fullband-melgan training using identical audio params (_not validated with CheckSpectrograms notebook_)
```
"audio": {
        "fft_size": 1024,
        "win_length": 1024,
        "hop_length": 256,
        "frame_length_ms": null,
        "frame_shift_ms": null,
        "sample_rate": 22050,
        "preemphasis": 0.0,
        "ref_level_db": 20,
        "do_trim_silence": true,
        "trim_db": 50,
        "num_mels": 80,
        "mel_fmin": 50.0,
        "mel_fmax": 8000.0,
        "spec_gain": 1.0,
        "signal_norm": true,
        "min_level_db": -100,
        "symmetric_norm": true,
        "max_norm": 4.0,
        "clip_norm": true,
        "stats_path": "/home/thorsten/___prj/tts/models/taco2/thorsten-dca/spec-stats.npy"
    },
```

Thanks in advance :-)

**Comment by erogol** [comment]
You can try training the vocoder using spectrograms extracted from your tacotron model.

You can also check how model works with spectrograms from original audio files to see if something is wrong out of the training.

**Reply by thorstenMueller** [reply]
Thanks @erogol .
Based on your reply and after chatting with @SanjaESC i'll try to finetune my fullband-melgan model (425k) with extracted spectrograms from my taco2 dca model. Spectrograms exported using https://github.com/coqui-ai/TTS/blob/dev/notebooks/ExtractTTSpectrogram.ipynb

**Comment by thorstenMueller** [comment]
I've extracted TTS spectrograms from my pretrained taco2 dca model and put the folder of .npy files in config as feature_path. Restoring training brings a mismatch on mel_shapes. Audio parameters are matching. It's the first try with "npy/features".

What did i do wrong?

```
> EPOCH: 0/10000

 > TRAINING (2021-04-14 17:39:29)
 ! Run is removed from /home/thorsten/___prj/tts/models/vocoder/FullBandMelGan/output/thorsten-dca-fullband-melgan-April-14-2021_05+39PM-a669a49
Traceback (most recent call last):
  File "./train_vocoder_gan.py", line 597, in <module>
    main(args)
  File "./train_vocoder_gan.py", line 568, in main
    epoch)
  File "./train_vocoder_gan.py", line 100, in train
    for num_iter, data in enumerate(data_loader):
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 517, in __next__
    data = self._next_data()
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 1199, in _next_data
    return self._process_data(data)
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 1225, in _process_data
    data.reraise()
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/torch/_utils.py", line 429, in reraise
    raise self.exc_type(msg)
AssertionError: Caught AssertionError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/torch/utils/data/_utils/worker.py", line 202, in _worker_loop
    data = fetcher.fetch(index)
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/TTS-0.0.11-py3.6-linux-aarch64.egg/TTS/vocoder/datasets/gan_dataset.py", line 70, in __getitem__
    item1 = self.load_item(idx)
  File "/home/thorsten/___prj/tts/models/release/lib/python3.6/site-packages/TTS-0.0.11-py3.6-linux-aarch64.egg/TTS/vocoder/datasets/gan_dataset.py", line 119, in load_item
    assert mel.shape[-1] * self.hop_len == audio.shape[-1], f' [!] {mel.shape[-1] * self.hop_len} vs {audio.shape[-1]}'
AssertionError:  [!] 65280 vs 65063
```

**Reply by thorstenMueller** [reply]
I trained fullband-melgan from step 0 to 425k with a reduced dataset (8k recordings) and restored it with full dataset (22k recordings, npy files). Might this be a problem?

**Reply by thorstenMueller** [reply]
@SanjaESC 🥇 solved it 👍 .
I had to switch do_trim_silence to "false". Even i've trained my taco2 models with this param set to "true" but when using it with feature_path is has to be set to false.
